src/map.rs

```diff
@@ -265,12 +260,19 @@ impl<K: Key, V: Val<A>, A: Actor> Map<K, V, A> {
         }
     }
 
-    /// Update a value under some key, if the key is not present in the map,
-    /// the updater will be given the result of V::default().
-    pub fn update<F, I>(&self, key: I, ctx: AddCtx<A>, f: F) -> Op<K, V, A>
+    /// Update a value under some key.
+    ///
+    /// If the key is not present in the map, the updater will be given the
+    /// result of `V::default()`. The `default` value is used to ensure
+    /// eventual consistency since our `Map`'s values are CRDTs themselves.
+    ///
+    /// The `impl Into<K>` bound provides a nice way of providing an input key that
+    /// can easily convert to the `Map`'s key. For example, we can call this function
+    /// with `"hello": &str` and it can be converted to `String`.
+    pub fn update<F>(&self, key: impl Into<K>, ctx: AddCtx<A>, f: F) -> Op<K, V, A>
     where
         F: FnOnce(&V, AddCtx<A>) -> V::Op,
-        I: Into<K>,
+        V: Default,
```

Comment:
Generally, I don't think it's useful to create an instance of `Map<K, V, A>` for a `V` that is not `Default`. All interesting behavior relies on their being a default V we can use.

Lets move this as a impl level constraint just to avoid some frustration from users who are able to create an instance but are then not able to use the instance to actually modify the Map.

ie.
```rust
impl<.., V: Val<A> + Default, ..> Map<..> {
    ...
}